File: DoConnect_Final/src/main/java/com/doconnect/service/interfaces/AuthService.java
package com.doconnect.service.interfaces;

import com.doconnect.dto.auth.AuthResponseDTO;
import com.doconnect.dto.auth.LoginRequestDTO;
import com.doconnect.dto.auth.RegisterRequestDTO;

public interface AuthService { 

    AuthResponseDTO register( 

            RegisterRequestDTO request); 

    AuthResponseDTO login( 

            LoginRequestDTO request); 

} 

File: doconnect_frontend_final/src/services/authService.js
/** 

 * Auth Service 

 * Register and login API calls 

 */ 

import axiosInstance from '../api/axiosConfig'; 
const authService = { 
  register: async (userData) => { 
    const response = await axiosInstance 

              .post('/auth/register', userData); 
    return response.data; 

  }, 
 login: async (credentials) => {
  const response = await axiosInstance.post('/auth/login', credentials);

  // ✅ Keep BOTH formats so LoginPage doesn't break
  return {
    ...response.data,          // keeps existing structure (status, message, etc.)
    token: response.data.data?.token,
    role: response.data.data?.role,
    username: response.data.data?.username,
    email: response.data.data?.email,
  };
},

}; 
export default authService; 